src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js

```diff
@@ -10,26 +10,56 @@ import EntitySelectionType from 'src/pages/workspaces/workspace/workflows/Entity
 
 
 export default ajaxCaller(class LaunchAnalysisModal extends Component {
+  /**
+   * Verify that the user's pet service account (used while running workflows) will be able to
+   * access the workspace bucket
+   *
+   * @returns {Promise<boolean>}
+   */
+  async preFlightBucketAccess() {
+    const { workspaceId: { namespace, name }, accessLevel, bucketName, ajax: { Workspaces } } = this.props
+
+    try {
+      await Workspaces.workspace(namespace, name).checkBucketAccess(bucketName, accessLevel)
+      return true
+    } catch (error) {
+      // Could check error.requesterPaysError here but for this purpose it really doesn't matter what the error was.
+      return false
+    }
+  }
+
+  async componentDidMount() {
+    this.setState({ checkingBucketAccess: true, message: 'Checking bucket access...' })
+    const hasBucketAccess = await this.preFlightBucketAccess()
```

Comment:
What about putting this check in the `doLaunch` function? That code path already has a notion of multiple steps, and potential errors leading to terminal error states (see `launchErrror`), so we might be able to leverage existing state instead of creating special cases for this check.